sky/shell/platform/ios/semantics_listener_impl.mm

```diff
@@ -0,0 +1,245 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/shell/platform/ios/semantics_listener_impl.h"
+
+namespace sky {
+namespace shell {
+
+SemanticsListenerImpl::SemanticsListenerImpl(
+    FlutterView* view,
+    semantics::SemanticsServerPtr semanticsServer)
+    : _view(view), _binding(this) {
+  mojo::InterfaceHandle<semantics::SemanticsListener> listener;
+  _binding.Bind(&listener);
+  semanticsServer->AddSemanticsListener(listener.Pass());
+}
+
+void SemanticsListenerImpl::UpdateSemanticsTree(
+    mojo::Array<semantics::SemanticsNodePtr> nodes) {
+  for (const semantics::SemanticsNodePtr& node : nodes) {
+    updateSemanticsNode(node);
+  }
+  NSArray* accessibleElements = createAccessibleElements();
+  [_view updateAccessibleElements:accessibleElements];
+  [accessibleElements release];
+}
+
+SemanticsListenerImpl::AccessibilityNode*
+SemanticsListenerImpl::updateSemanticsNode(
+    const semantics::SemanticsNodePtr& node) {
+  AccessibilityNode* persistentNode;
+  NodeMapIterator iter = _nodes.find(node->id);
+  if (iter == _nodes.end()) {
+    persistentNode = new AccessibilityNode(this, node);
+    _nodes.insert(NodeMap::value_type(node->id, persistentNode));
+  } else {
+    persistentNode = iter->second;
+    persistentNode->update(node);
+  }
+  assert(persistentNode != NULL);
+  return persistentNode;
+}
+
+void SemanticsListenerImpl::removePersistentNode(AccessibilityNode* node) {
+  assert(_nodes.find(node->id) != _nodes.end());
+  assert(_nodes.at(node->id)->parent == NULL);
+  _nodes.erase(node->id);
+  for (AccessibilityNode* child : node->children) {
+    removePersistentNode(child);
+  }
+}
+
+NSArray* SemanticsListenerImpl::createAccessibleElements() {
```

Comment:
Though you have correctly named the method to indicate transfer-out ownership of the `NSArray` instance (`init`, `create`, `new` have the same meaning by convention), its better to declare the method with `NS_RETURNS_RETAINED` (or whatever `__attribute__` that ends up resolving to). This is similar to the `ownership_returns` and `ownership_takes` decorations for `malloc`.